packages/clerk-js/src/utils/captcha.ts

```diff
@@ -112,29 +112,29 @@ export const getCaptchaToken = async (captchaOptions: {
     return div;
   };
 
-  if (invisibleWidget) {
-    widgetDiv = createInvisibleDOMElement();
-  } else {
-    const visibleDiv = document.getElementById(CAPTCHA_ELEMENT_ID);
-    if (visibleDiv) {
-      visibleDiv.style.display = 'block';
-      widgetDiv = visibleDiv;
-    } else {
-      console.error('Captcha DOM element not found. Using invisible captcha widget.');
-      widgetDiv = createInvisibleDOMElement();
-      invisibleWidget = true;
-      turnstileSiteKey = invisibleSiteKey;
-    }
-  }
-
   const captcha = await loadCaptcha(scriptUrl);
   let retries = 0;
   const errorCodes: (string | number)[] = [];
 
   const handleCaptchaTokenGeneration = (): Promise<[string, string]> => {
     return new Promise((resolve, reject) => {
       try {
-        const id = captcha.render(invisibleWidget ? `.${CAPTCHA_INVISIBLE_CLASSNAME}` : `#${CAPTCHA_ELEMENT_ID}`, {
+        if (isInvisibleWidget) {
+          widgetDiv = createInvisibleDOMElement();
+        } else {
+          const visibleDiv = document.getElementById(CAPTCHA_ELEMENT_ID);
+          if (visibleDiv) {
+            visibleDiv.style.display = 'block';
+            widgetDiv = visibleDiv;
+          } else {
+            console.error('Captcha DOM element not found. Using invisible captcha widget.');
+            widgetDiv = createInvisibleDOMElement();
+            isInvisibleWidget = true;
+            turnstileSiteKey = invisibleSiteKey;
+          }
+        }
```

Comment:
❓ It's not obvious to me, why moving this code is beneficial, could you add more details ?

Reply:
@panteliselef added more context in the description of the PR